### PR TITLE
WIP: HTML trimming choice

### PR DESCRIPTION
**Pull request by msvana**

### CODE REVIEW

File: ai_powered_qa/custom_plugins/website_explorer.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+class WebsiteExplorer(Plugin):
+    name: str = "WebsiteExplorer"
+    _client: OpenAI = PrivateAttr()
+
+    def __init__(self, **data):
+        super().__init__(**data)
+        self._client = OpenAI()
+
+    @tool
+    def find_element_to_perform_action(self, action_description: str, html: str):
```
Comment (vnawrath) [inline ai_powered_qa/custom_plugins/website_explorer.py:16]: Interesting concept. Is the idea that the agent only passes a snippet of the overall HTML, which is a good candidate for the selector? I think we might get even more out of it if the agent only receives the description of the page, and this plugin gets the HTML through a side channel.
That way, the agent only has the task of figuring out the suitable action in plain english, and isn't distracted by the cognitive load of the HTML. And the tool costs less token, as we would only be passing the decription of the action I want to take, and not the HTML.

File: tests/test_playwright.py [hunk 2]
```diff
@@ -27,13 +27,10 @@ def test_playwright_click_incorrect():
...
-        "click_element", selector="a[href='/something-that-doesnt-exist']", index=0
-    )
-    assert (
-        response
-        == "Unable to click on element 'a[href='/something-that-doesnt-exist'] >> visible=1 >> nth=0' as it does not exist"
-    )
+    selector = "a[href='/something-that-doesnt-exist']"
+    response = plugin.call_tool("click_element", selector=selector)
+    expected = f"Unable to click on element '{selector}'"
+    assert response == expected
```
Comment (vnawrath) [inline tests/test_playwright.py:33]: I probably broke some of these in my PR, sorry about that. We might consider running the tests as a GH action.

File: web_ui_next.py [hunk 3]
```diff
@@ -13,11 +21,24 @@
...
+    "PlaywrightPluginOnlyVisible": PlaywrightPluginOnlyVisible,
+}
+
+agent_store = AgentStore("agents", name_to_plugin_class=NAME_TO_PLUGIN_CLASS)
+
+
+st.write(
+    """
+    <style>
+        pre:has(code.language-html) {
```
Comment (vnawrath) [inline web_ui_next.py:36]: 👍

**Review by vnawrath — APPROVED** [review]
Love it!

File: tests/test_playwright.py (hunk 2 above)
Comment (msvana) [reply tests/test_playwright.py:33]: Agreed ..

Comment (msvana) [reply tests/test_playwright.py:33]: All tests work on my PC :D

**Pull request merged**